### ISSUE TITLE
Handling multi-ipmb channels

### ISSUE THREAD

**Issue by warp5tw** [openbmc/ipmbbridge#3]
Hi Ed:

This is Tyrone from Nuvoton. Thank you for your prompt feedback about ipmbd in OpenBMC mailing list.

I'm using the ipmbbridge, Nuvoton NPCM7XX platform and the i2c slave mqueue driver to do an IPMB test.

OpenBMC software stack is running on Nuvoton NPCM7XX platform.

May I know if ipmbbridge is going to support "multi-ipmb channels"?

That is, when an IPMI "send message" command reaches to BMC, the command's request data contains a field "channel number to send message to.".

So if a board with a BMC contains multiple IPMB headers/channels, the BMC needs to know to which physical IPMB channel it would send the real command which is contained within the "send message" command by identifying the "channel number" field in the "send message" command's request data.

In other words, BMC has a one-for-one mapping between the "channel number" field in the "send message" command's request data and the physical IPMB channels.

Is ipmbbridge is going to support it?

Thank you.

Regards,
Tyrone

**Comment by edtanous** [comment]
The intent long term is to support multiple IPMB channels.  There is work going on within IPMI to refactor how channel numbering is accomplished.   Today, the ipmi daemon assumes that there is only a single outgoing channel, and that is assigned the IPMB channel number for the Send Message command (which isn't yet implemented).

ipmibridge (once I've posted the refactor) will support multiple lanes of IPMB, but we don't have a timeframe on the changes to IPMI host to make the channel number implementation more robust, although nothing in ipmbbridge should prevent it, as it's been tested over DBus directly.

In short, yes, we will support multiple lanes of IPMB requests initially, but may only support a single channel initially for IPMB bridging.
